enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,585 @@
+
+**Cluster Creator** Uses running Agents to create and grow clusters.
+
+#### Day 2 Add Node Virtualmedia Multicluster (add Remote Worker Node)
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke cluster.
+
+1. Infra Owner creates a BareMetalHost resource with a label that matches an InstallEnv selector.
+1. The Baremetal Agent Controller adds the discovery ISO URL to the BareMetalHost.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO on the BareMetalHost.
+1. The Agent starts up and reports back to the assisted service, which creates an Agent resource in the cluster. The Agent is labeled with the labels that were specified in the InstallEnv's Spec.
+1. The Agent's Role field in its spec is assigned the value "master" or "worker" if a corresponding label was present on its BareMetalHost. (only "worker" is supported for now on day 2)
+1. The Agent is marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. The Baremetal Agent Controller adds inspection data found on the Agent's Status to the BareMetalHost.
+1. When the agent is in a ready state, installation of that host begins.
+
+#### Create Cluster
+
+This scenario takes place on a hub cluster where hive and possibly RHACM are
+present. This scenario does not include centralized machine management.
+
+1. Infra Owner creates an InstallEnv resource. It can include fields such as egress proxy, NTP server, ssh public key, ... In particular it includes an Agent label selector, and a separate field of labels that should be applied to Agents.
+1. Infra Owner creates BareMetalHost resources that include BMC credentials. They are labeled so that they match the selctor on the InstallEnv.
+1. A new controller, the Baremetal Agent Controller, sees the matching BareMetalHosts and boots them using the discovery ISO URL found in the InstallEnv's status.
+1. The Agent starts up on each host and reports back to the assisted service, which creates an Agent resource in the cluster. The Agent is labeled with the labels that were specified in the InstallEnv's Spec.
+1. The Agent's Role field in its spec is assigned the value "master" or "worker" if a corresponding label was present on its BareMetalHost.
+1. The Agent is marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. Cluster Creator creates a ClusterDeployment describing a new cluster. It describes how many master and worker agents to expect. It also includes a label selector to match Agent resources that should be part of the cluster.
```

Comment:
Suggest "It describes the exact number of control plane agents, and the minimum number of worker agents required."